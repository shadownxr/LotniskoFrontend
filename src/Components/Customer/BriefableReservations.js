import React, { useState,useEffect } from "react";
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import SearchButton from './SearchReservationButton';
import BriefableReservationsList from './BriefableReservationsList';
import Cookie from 'react-cookies';

const StyledTableCell = withStyles((theme) => ({
    head: {
      backgroundColor: theme.palette.common.black,
      color: theme.palette.common.white,
    },
    body: {
      fontSize: 14,
    },
  }))(TableCell);
  
  const StyledTableRow = withStyles((theme) => ({
    root: {
      '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }))(TableRow);

const useStyles = makeStyles({
    table: {
      minWidth: 650,
    },
});

/**
 * Renders array with briefable reservations
 * @param {refresh,accountData} props 
 */
export default function BriefableReservations(props){
    const [reservations,setReservations] = useState([]);
    const [refresh,setRefresh] = useState(props.refresh);

    useEffect(() => {
      fetchReservations();
    },[]);

    useEffect(() => {
      if(refresh === true){
        fetchReservations();
        setRefresh(false);
      }
    },[refresh]);

    const fetchReservations = () => {
        let payload = {
          "userID": props.accountData.id,
        }

      const url = "https://localhost:8443/api/tickets/listByClientID";

      const options = {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'Authorization': 'Bearer ' + Cookie.load('userToken').token,
          },
          body: JSON.stringify(payload),
      };

      fetch(url, options)
        .then(response => response.json())
        .then(result => {
          setReservations(result);
        });
    }

    return(
        <div>
            <TableContainer component={Paper}>
                <Table className={useStyles().table} aria-label="simple table">
                    <TableHead>
                        <StyledTableRow>
                            <StyledTableCell align="center" colSpan={7}>Check-in</StyledTableCell>
                        </StyledTableRow>
                        <StyledTableRow>
                            <StyledTableCell align="center" colSpan={2}>From</StyledTableCell>
                            <StyledTableCell align="center" colSpan={2}>To</StyledTableCell>
                            <StyledTableCell align="center">Departure</StyledTableCell>
                            <StyledTableCell align="center">Class</StyledTableCell>
                            <StyledTableCell align="center"><SearchButton /></StyledTableCell>
                        </StyledTableRow>
                    </TableHead>
                    <BriefableReservationsList customerReservationsData={reservations} reservation={(reservation) => {props.reservation(reservation)}} screen={(screen) => {props.screen(screen)}}/>
                </Table>
            </TableContainer>
        </div>
    )
}