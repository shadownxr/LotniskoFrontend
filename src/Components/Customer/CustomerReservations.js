import React from "react";
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import SearchButton from './SearchReservationButton';
import CustomerReservationsList from './CustomerReservationsList';

const StyledTableCell = withStyles((theme) => ({
    head: {
      backgroundColor: theme.palette.common.black,
      color: theme.palette.common.white,
    },
    body: {
      fontSize: 14,
    },
  }))(TableCell);
  
  const StyledTableRow = withStyles((theme) => ({
    root: {
      '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }))(TableRow);

const useStyles = makeStyles({
    table: {
      minWidth: 650,
    },
});

export default function CustomerReservations(){
    const reservations = {

    }
    return(
        <div>
            <TableContainer component={Paper}>
                <Table className={useStyles().table} aria-label="simple table">
                    <TableHead>
                        <StyledTableRow>
                            <StyledTableCell align="center" colSpan={7}>Moje rezerwacje</StyledTableCell>
                        </StyledTableRow>
                        <StyledTableRow>
                            <StyledTableCell align="center">Nr</StyledTableCell>
                            <StyledTableCell align="center">Z</StyledTableCell>
                            <StyledTableCell align="center">Do</StyledTableCell>
                            <StyledTableCell align="center">Data wylotu</StyledTableCell>
                            <StyledTableCell align="center">Klasa</StyledTableCell>
                            <StyledTableCell align="center">Cena</StyledTableCell>
                            <StyledTableCell align="center"><SearchButton /></StyledTableCell>
                        </StyledTableRow>
                    </TableHead>
                    <CustomerReservationsList customerReservationsData={reservations}/>
                </Table>
            </TableContainer>
        </div>
    )
}