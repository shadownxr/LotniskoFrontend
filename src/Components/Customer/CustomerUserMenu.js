import React from 'react';
import Button from '@material-ui/core/button';

export default function CustomerUserMenu(props){
    const handleMenuChoice = (choice) => {
        props.menuChoice(choice);
    }

    return(
        <div>
            <div className="UserMenu">
                <Button style={{color: '#fbfd8a'}} onClick={() => handleMenuChoice("View Flights")}>FLIGHTS</Button>
                <Button style={{color: '#fbfd8a'}}  onClick={() => handleMenuChoice("Customer Reservations")}>MY TICKETS</Button>
                <Button style={{color: '#fbfd8a'}}  onClick={() => handleMenuChoice("Online Briefing")}>CHECK IN</Button>
            </div>
        </div>
    )
}