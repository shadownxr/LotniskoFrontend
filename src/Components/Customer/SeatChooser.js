import React, { useState, useEffect } from 'react';
import Cookie from 'react-cookies';
import Autocomplete from '@material-ui/lab/Autocomplete';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

/**
 * Form to choose seat in a plane
 * @param {reservation,seat,screen} props 
 */
export default function SeatChooser(props){
    const [value,setValue] = useState("");
    const [seats,setSeats] = useState([]);

    useEffect(() => {
        fetchFreeSeats();
    },[])

    const handleNext = () => {
      props.seat(value);
      props.screen(4);
    }

    const fetchFreeSeats = () => {
        let payload = {
            "flightId": props.reservation.flightID.id,
            "className": props.reservation.className
          }
  
        const url = "https://localhost:8443/api/flights/getFreeSeats";
  
        const options = {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json',
              'Accept': 'application/json',
              'Authorization': 'Bearer ' + Cookie.load('userToken').token,
            },
            body: JSON.stringify(payload),
        };
  
        fetch(url, options)
          .then(response => response.json())
          .then(result => {
            setSeats(result.map((number) => number.toString()));
          });
      }

    return(
        <div>
            <div style={{display:"flex",flexDirection:"column",backgroundColor:"white",justifyContent:"center",alignItems:"center"}}>
                <div style={{flex:"1"}}>Wybierz miejsce</div>
                  <div style={{flex:"2"}}>
                    <Autocomplete
                      onChange={(event, newValue) => {
                        setValue(newValue);
                      }}
                      id="Seats"
                      options={seats}
                      style={{ width: 300 }}
                      renderInput={(params) => <TextField {...params} label={props.reservation.className} variant="outlined" />}
                    />
                  </div>
                  <div style={{flex:"3"}}><Button onClick={() => handleNext()}>Dalej</Button></div>
            </div>
        </div>
    )
}