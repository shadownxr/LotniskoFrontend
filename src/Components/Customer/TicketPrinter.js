import React from 'react';
import Button from '@material-ui/core/Button';
import jsPDF from 'jspdf';
import Cookie from 'react-cookies';

/**
 * Creates pdf ticket and checks that breiefing was compleated
 * @param {ticketData,reservation,seat} props 
 */
export default function TicketPrinter(props){
    const jsPdfGenerator = () => {
        let doc = new jsPDF('p','pt');
        doc.text(20,20,"Name: "+props.ticketData.name);
        doc.text(20,35,"Surname: "+props.ticketData.surename);
        doc.text(20,50,"DOB: "+new Date(props.ticketData.birthDate).toLocaleDateString());
        doc.text(20,65,"Country: "+props.ticketData.country);
        doc.text(20,80,"City: "+props.ticketData.address);
        doc.text(20,95,"Street: "+props.ticketData.street);
        doc.text(20,110,"House number:" +props.ticketData.homeNr);

        doc.text(300,20,"From: "+props.reservation.flightID.sapid.airportName+" "+props.reservation.flightID.sapid.cityName);
        doc.text(300,35,"To: "+props.reservation.flightID.dapid.airportName+" "+props.reservation.flightID.dapid.cityName);
        doc.text(300,50,"Departure: "+new Date(props.reservation.flightID.startDate).toLocaleDateString());
        doc.text(300,65,"Arrival: "+new Date(props.reservation.flightID.endDate).toLocaleDateString());
        doc.text(300,80,"Plane: "+props.reservation.flightID.planeID.planeName);
        doc.text(300,95,"Seat: "+props.seat);

        doc.save('Bilet.pdf');
    }

    const fetchCheckIn = () => {
        let payload = {
            "ticketID": props.reservation.id,
            "seat": props.seat
          }
  
        const url = "https://localhost:8443/api/tickets/checkIn";
  
        const options = {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json',
              'Accept': 'application/json',
              'Authorization': 'Bearer ' + Cookie.load('userToken').token,
            },
            body: JSON.stringify(payload),
        };
  
        fetch(url, options)
          .then(response => response.json())
          .then(result => {
          });
    }

    const handleTicket = () => {
        fetchCheckIn();
        jsPdfGenerator();
        props.refresh(true);
        props.screen(1);
    }

    return(
        <div style={{backgroundColor:"white"}}>
            <h1>Check-in</h1>
            <div style={{display:"flex",flexDirection:"row",justifyContent:"center"}}>
                <div style={{flex:"1",alignContent:"center",maxWidth:"50%"}}>
                    Name: {props.ticketData.name} <br/>
                    Surname: {props.ticketData.surename} <br/>
                    DOB: {new Date(props.ticketData.birthDate).toLocaleDateString()} <br/>
                    Country: {props.ticketData.country} <br/>
                    City: {props.ticketData.address} <br/>
                    Street: {props.ticketData.street} <br/>
                    House number: {props.ticketData.homeNr} <br/>
                </div>
                <div style={{flex:"2",alignContent:"center",maxWidth:"50%"}}>
                    From: {props.reservation.flightID.sapid.airportName} {props.reservation.flightID.sapid.cityName}<br/>
                    To: {props.reservation.flightID.dapid.airportName} {props.reservation.flightID.dapid.cityName}<br/>
                    Departure: {new Date(props.reservation.flightID.startDate).toLocaleDateString()}<br/>
                    Arrival: {new Date(props.reservation.flightID.endDate).toLocaleDateString()}<br/>
                    Plane: {props.reservation.flightID.planeID.planeName}<br/>
                    Seat: {props.seat}<br/>
                </div>
            </div>
            <Button onClick={() => handleTicket()}>Get the ticket</Button>
        </div>
    )
}