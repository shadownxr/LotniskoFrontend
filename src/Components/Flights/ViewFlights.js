import React, { useState, useEffect } from 'react';
import ViewFlightsList from './ViewFlightsList';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import SearchButton from './SearchFlightButton';


const StyledTableCell = withStyles((theme) => ({
    head: {
      backgroundColor: theme.palette.common.black,
      color: theme.palette.common.white,
    },
    body: {
      fontSize: 14,
    },
  }))(TableCell);
  
  const StyledTableRow = withStyles((theme) => ({
    root: {
      '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }))(TableRow);

const useStyles = makeStyles({
    table: {
      minWidth: 650,
    },
});

export default function ViewFlights(props){
    const [flights,setFlights] = useState([]);

    useEffect(() => {
      fetchFlights();
    },[]);

    const fetchFlights = () => {
      const url = "http://localhost:8080/api/flights/list";

      let options = {
        method: 'GET',
      }

      fetch(url, options)
        .then(response => response.json())
        .then(result => {
          console.log(result);
          setFlights(result);
        });
    }

    return(
        <div>
            <TableContainer component={Paper}>
                <Table className={useStyles().table} aria-label="simple table">
                    <TableHead>
                        <StyledTableRow>
                            <StyledTableCell align="center" colSpan={7}>Przeglądarka lotów</StyledTableCell>
                        </StyledTableRow>
                        <StyledTableRow>
                            <StyledTableCell align="center">Nr</StyledTableCell>
                            <StyledTableCell align="center">Z</StyledTableCell>
                            <StyledTableCell align="center">Do</StyledTableCell>
                            <StyledTableCell align="center">Data wylotu</StyledTableCell>
                            <StyledTableCell align="center">Klasa</StyledTableCell>
                            <StyledTableCell align="center">Cena</StyledTableCell>
                            <StyledTableCell align="center"><SearchButton /></StyledTableCell>
                        </StyledTableRow>
                    </TableHead>
                    <ViewFlightsList flightsData={flights} accountData={props.accountData}/>
                </Table>
            </TableContainer>
        </div>
    )
}