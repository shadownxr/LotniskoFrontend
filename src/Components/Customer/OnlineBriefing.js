import React, { useState } from 'react';
import BriefableReservations from './BriefableReservations';
import UserDataForm from './UserDataForm';
import TicketPrinter from './TicketPrinter';
import SeatChooser from './SeatChooser';

export default function OnlineBriefing(props){
    const [briefingScreen, setBriefingScreen] = useState(1);
    const [reservation, setReservation] = useState([]);
    const [ticketData, setTicketData] = useState({});
    const [seat, setSeat] = useState("");
    const [refresh, setRefresh] = useState(false);

    const briefing = () => {
        switch(briefingScreen){
            case 1:
                return <BriefableReservations accountData={props.accountData} refresh={refresh} reservation={(reservation) => {setReservation(reservation)}} screen={(screen) => {setBriefingScreen(screen)}}/>;
            case 2:
                return <UserDataForm screen={(screen) => {setBriefingScreen(screen)}} ticketData={(ticketData) => {setTicketData(ticketData)}}/>;
            case 3:
                return <SeatChooser reservation={reservation} seat={(seat) => {setSeat(seat)}} screen={(screen) => {setBriefingScreen(screen)}}/>;
            case 4:
                return <TicketPrinter reservation={reservation} seat={seat} ticketData={ticketData} refresh={(refresh) => {setRefresh(refresh)}} screen={(screen) => {setBriefingScreen(screen)}}/>;
            default:
                return <BriefableReservations accountData={props.accountData} reservation={(reservation) => {setReservation(reservation)}} screen={(screen) => {setBriefingScreen(screen)}} refresh={(refresh) => {setRefresh(refresh)}}/>;
        }
    }

    return(
        <div>{briefing()}</div>
    )
}