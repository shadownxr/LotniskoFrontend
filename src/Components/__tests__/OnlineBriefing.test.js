import React from 'react';
import { shallow } from 'enzyme';

import ViewFlights from "../Flights/ViewFlights";
import OnlineBriefing from "../Customer/OnlineBriefing";
import CustomerReservations from "../Customer/CustomerReservations";
import CustomerUserMenu from "../Customer/CustomerUserMenu";
import BriefableReservations from "../Customer/BriefableReservations";
import SeatChooser from "../Customer/SeatChooser";
import TicketPrinter from "../Customer/TicketPrinter";

const Test = {flightID: {
        sapid: {
            airportName: "testOD",
            cityName: "OD",
        },
        dapid: {
            airportName: "testDO",
            cityName: "DO",
        },
        startDate: "2020-10-02",
    },
    className: "Economic"};

const Test2 = {flightID: {
        sapid: {
            airportName: "testOD2",
            cityName: "OD2",
        },
        dapid: {
            airportName: "testDO2",
            cityName: "DO2",
        },
        startDate: "2020-10-03",
    },
    className: "Economic"};

const reservation = [Test, Test2];
const accountData = {username: "Admin", email: "Admin@Example.com"};

it('renders without crashing', () => {
    shallow(<OnlineBriefing accountData={accountData}/>);
});

describe('CustomerUserMenu test',() =>{
    it('Renders when loaded',() =>{
        const wrapper = shallow(<OnlineBriefing accountData={accountData} briefingScreen={1}/>);
        expect(wrapper.find(<BriefableReservations accountData={accountData} reservation={reservation} />)).toMatchSnapshot();
    })

    it('Renders when loaded',() =>{
        const wrapper = shallow(<OnlineBriefing accountData={accountData} briefingScreen={3}/>);
        expect(wrapper.find(<SeatChooser reservation={Test} seat={1} />)).toMatchSnapshot();
    })

    it('Renders when loaded',() =>{
        const wrapper = shallow(<OnlineBriefing accountData={accountData} briefingScreen={4} />);
        expect(wrapper.find(<TicketPrinter reservation={Test} seat={1} ticketData={"2020-10-01"} />)).toMatchSnapshot();
    })

    it('Renders when loaded',() =>{
        const wrapper = shallow(<OnlineBriefing accountData={accountData} />);
        expect(wrapper.find(<BriefableReservations accountData={accountData} reservation={reservation} />)).toMatchSnapshot();
    })

})
