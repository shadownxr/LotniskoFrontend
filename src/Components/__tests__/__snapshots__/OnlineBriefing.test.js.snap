// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerUserMenu test Renders when loaded 1`] = `ShallowWrapper {}`;

exports[`CustomerUserMenu test Renders when loaded 2`] = `ShallowWrapper {}`;

exports[`CustomerUserMenu test Renders when loaded 3`] = `ShallowWrapper {}`;

exports[`CustomerUserMenu test Renders when loaded 4`] = `ShallowWrapper {}`;
